app/src/main/java/ch/epfl/polychef/fragments/UserProfileFragment.java

```diff
@@ -109,6 +109,10 @@ public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newStat
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
+        if(!hostActivity.isOnline()){
+            Toast.makeText(hostActivity, "You are not connected to the internet", Toast.LENGTH_SHORT).show();
+        }
+
```

Comment:
What about
```
if(dynamicRecipeList.isEmpty()){
            getNextRecipes();
}
```
at the bottom of this method? Shouldn't you prevent `getNRecipes` from being called?

Reply:
Okay, I thought the queries were the reason the app crashed, if that's not the case then it's fine to let the queries through